### PR TITLE
Task edit content

### PR DESCRIPTION
**Pull request by aabdiaziz**

### CODE REVIEW

File: php/aboutMeFunctions.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+/*
+* This Function Fetches Data from the database and returns the query as an associative array.
+*
+* @param the $db is to input the new instance of PDO database.
+*
+*
+* @return Assoc array which contains results.
+*
+*/
+function fetchData($db) {
```
Comment (mattap86) [inline php/aboutMeFunctions.php]: type hint PDO

File: php/aboutMe.php [hunk 2]
```diff
@@ -1,64 +1,10 @@
 <?php
 
-$db = new PDO('mysql:host=127.0.0.1;dbname=Portfolio','root');
-//fetching everything from database [about me section]
-
-    /*
-    * This Function Fetches Data from the database and returns the query as an associative array.
-    *
-    * @param the $db is to input the new instance of PDO database.
```
Comment (mattap86) [inline php/aboutMe.php]: @param missing datatype before variable.

File: php/aboutMe.php [hunk 3]
```diff
@@ -1,64 +1,10 @@
...
-    *
-    *
-    * @return Assoc array which contains results.
```
Comment (mattap86) [inline php/aboutMe.php]: @return missing variable after datatype and doesn't explain what $results represents

File: php/aboutMeFunctions.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+* This Function Fetches Data from the database and returns the query as an associative array.
+*
+* @param the $db is to input the new instance of PDO database.
```
Comment (mattap86) [inline php/aboutMeFunctions.php]: @param missing datatype before variable instead of 'the'

File: php/aboutMeFunctions.php [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+*
+*
+* @return Assoc array which contains results.
```
Comment (mattap86) [inline php/aboutMeFunctions.php]: @return missing variable after datatype and does not explain what variable represents

File: php/aboutMeFunctions.php [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+    $query->execute();
+    $results = $query->fetch();
+    return $results;
+}
+
+/*
+* This Function takes the results and outputs the content field in the database.
+*
+* @param the Parameter it takes will be $results from before.
```
Comment (mattap86) [inline php/aboutMeFunctions.php]: @param missing datatype and variable before description.
Description vague.
Suggestion: "@param array $results represents the associative array retrieved from the fetchData function"

File: php/aboutMeFunctions.php [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+*
+*
+* @return returns the value in a string format.
```
Comment (mattap86) [inline php/aboutMeFunctions.php]: @return missing datatype and variable.
2 return options in function should have 2 @return descriptions.
Suggestion: @return string returns error message.

File: php/aboutMeFunctions.php [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+function fetchContent(array $results):string{
+    if(array_key_exists('content', $results)){
+        return $results["content"];
+    } else {
+        return 'error';
+    }
+}
+/*
+* This Function takes the results and outputs the title field in the database.
```
Comment (mattap86) [inline php/aboutMeFunctions.php]: same comments as with above function

File: php/dbConn.php [hunk 9]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+
+/**
+ * This function standardises db connection for the whole architecture
+ *
+ * @return PDO
```
Comment (mattap86) [inline php/dbConn.php]: no description of return

File: php/editContent.php [hunk 10]
```diff
@@ -1 +1,32 @@
...
+    <meta charset="UTF-8">
+    <title>Abdi's Portfolio</title>
+    <link rel="stylesheet" type="text/css" href="editContent.css">
+</head>
+<body>
+<div class="content">
+    <h1>Edit About Me</h1>
+    <form action="update.php" method="post" class="formContent">
+        Edit Title:<input type="text" name="title" class="titleContent" value="<?php echo $title; ?>">
+        <br>
```
Comment (mattap86) [inline php/editContent.php]: <br> tags have pulled up with me in a code review before so might want to lose them before final story code review.
lines 19,20,23-25

File: php/updateFunctions.php [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+require_once 'dbConn.php';
+
+$db = dbConn();
+
+
+
+    /*
+    * This Function Updates given fields into the database.
+    *
+    * @param the $db is to input the new instance of PDO database.
```
Comment (mattap86) [inline php/updateFunctions.php]: change the for PDO

File: php/updateFunctions.php [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+    * @param the $title contains the value of title to be inputted.
```
Comment (mattap86) [inline php/updateFunctions.php]: change the for string and swap content and title variables to read in order of input

File: php/updateFunctions.php [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+    function insertIntoDb($db,$title,$content){
+        $query = $db->prepare("UPDATE `about_content` SET `title_of_content`=:title, `content`=:content WHERE `about_content`.`id` = '1';");
+
+        $query->execute(['title'=>$title,'content'=>$content]);
+
+        return $query->execute();
+    }
+
+} else {
+    echo 'Error Please Enter A string Input';
```
Comment (mattap86) [inline php/updateFunctions.php]: don't know if this should be added to the docBlock as an alternative outcome of the function?

**Comment by mattap86** [comment]
Notes:
Couldn't see tests for functions on lines 33 & 50 in 'aboutMeFunctions.php' so was unable to review them.

Few <br> tags which I've been pulled up on in previous code reviews so may want to change them before final story review.

Changes:
Have noted the potential issues with docBlocks in the files changed tab, inline.

File: php/aboutMeFunctions.php (hunk 1 above)
Comment (aabdiaziz) [reply php/aboutMeFunctions.php]: done

File: php/aboutMeFunctions.php (hunk 4 above)
Comment (aabdiaziz) [reply php/aboutMeFunctions.php]: done

File: php/aboutMeFunctions.php (hunk 5 above)
Comment (aabdiaziz) [reply php/aboutMeFunctions.php]: done

File: php/aboutMeFunctions.php (hunk 6 above)
Comment (aabdiaziz) [reply php/aboutMeFunctions.php]: done

File: php/aboutMeFunctions.php (hunk 7 above)
Comment (aabdiaziz) [reply php/aboutMeFunctions.php]: done

File: php/aboutMeFunctions.php (hunk 8 above)
Comment (aabdiaziz) [reply php/aboutMeFunctions.php]: done

File: php/dbConn.php (hunk 9 above)
Comment (aabdiaziz) [reply php/dbConn.php]: done

File: php/editContent.php (hunk 10 above)
Comment (aabdiaziz) [reply php/editContent.php]: done but kept one

**Comment by Skylucas** [comment]
Okay looks good

**Review by Skylucas — APPROVED** [review]
All changes have been met from all sections.

**Pull request merged**